cmd/cert-checker/main.go

```diff
@@ -74,32 +70,47 @@ type certChecker struct {
 	clock        clock.Clock
 	rMu          *sync.Mutex
 	issuedReport report
+	checkPeriod  time.Duration
+	stats        statsd.Statter
 }
 
-func newChecker(saDbMap *gorp.DbMap, paDbMap *gorp.DbMap, clk clock.Clock, enforceWhitelist bool, challengeTypes map[string]bool) certChecker {
+func newChecker(saDbMap *gorp.DbMap, paDbMap *gorp.DbMap, clk clock.Clock, enforceWhitelist bool, challengeTypes map[string]bool, period time.Duration) certChecker {
 	pa, err := policy.New(paDbMap, enforceWhitelist, challengeTypes)
 	cmd.FailOnError(err, "Failed to create PA")
 	c := certChecker{
-		pa:    pa,
-		dbMap: saDbMap,
-		certs: make(chan core.Certificate, batchSize),
-		rMu:   new(sync.Mutex),
-		clock: clk,
+		pa:          pa,
+		dbMap:       saDbMap,
+		certs:       make(chan core.Certificate, batchSize),
+		rMu:         new(sync.Mutex),
+		clock:       clk,
+		checkPeriod: period,
 	}
 	c.issuedReport.Entries = make(map[string]reportEntry)
 
 	return c
 }
 
-func (c *certChecker) getCerts() error {
-	c.issuedReport.end = c.clock.Now()
-	c.issuedReport.begin = c.issuedReport.end.Add(-checkPeriod)
+const (
+	getCertsCountQuery = "SELECT count(*) FROM certificates WHERE issued >= :issued"
+	getCertsQuery      = "SELECT * FROM certificates WHERE issued >= :issued"
+)
 
+func (c *certChecker) getCerts(unexpiredOnly bool) error {
+	c.issuedReport.end = c.clock.Now()
+	c.issuedReport.begin = c.issuedReport.end.Add(-c.checkPeriod)
+
+	cq := getCertsCountQuery
+	args := map[string]interface{}{"issued": c.issuedReport.begin}
+	now := c.clock.Now()
+	if unexpiredOnly {
+		cq += " AND expires >= :now"
```

Comment:
Can we merge this clause and the one below into the overall query, and control them by setting now = 0 unless unexpiredOnly is true?